### DISCUSSION TITLE
What's exit code 255 meant to mean?

### DISCUSSION THREAD

**Discussion by mjr4077au** [PowerShell/PowerShell#21474] [Q&A]
I've had issues in the past where I've seen exit codes like `64`, meaning the provided command line was bad. I've got a situation now where I have otherwise reliable scripts exiting with `255`, and I'm not sure why.

Appreciate the below is from Windows PowerShell but I'm tied to it for now. This is literally all I receive out of a global transcript log when a script happens to fail in this manner.

```
**********************
Windows PowerShell transcript start
Start time: 20240416030247
Username: WORKGROUP\SYSTEM
RunAs User: WORKGROUP\SYSTEM
Configuration Name: 
Machine: SIPL-G5GDQ73 (Microsoft Windows NT 10.0.22631.0)
Host Application: C:\Windows\system32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -NoProfile -NonInteractive -WindowStyle Hidden -ExecutionPolicy Bypass -File C:\Program Files\WindowsPowerShell\Modules\TMLSTL.PSADT\Invoke-PsadtOperation.ps1 -Executable https://officecdn.microsoft.com/pr/wsus/setup.exe -DeploymentType Install -BlockingApps EXCEL,lync,lync99,MSACCESS,MSPUB,ONENOTE,ONENOTEM,OUTLOOK,POWERPNT,WINWORD -DeployMode NonInteractive -Arguments C:\Windows\TEMP\1363771850.xml -CurrentPath C:\Windows\IMECache\76c3c57b-e8c6-4086-9f06-4e64c2f6716d_3 -AppName Microsoft 365 Apps for enterprise
Process ID: 10420
PSVersion: 5.1.22621.2506
PSEdition: Desktop
PSCompatibleVersions: 1.0, 2.0, 3.0, 4.0, 5.0, 5.1.22621.2506
BuildVersion: 10.0.22621.2506
CLRVersion: 4.0.30319.42000
WSManStackVersion: 3.0
PSRemotingProtocolVersion: 2.3
SerializationVersion: 1.1.0.1
**********************

```

**Comment by SeeminglyScience** [comment]
You would likely have to ask either PSADT, or the vendor providing the setup file. In that example PSADT is likely checking the exit code of the installer and exiting with that error code if it's not 0. As far as I know PowerShell itself only ever exits with `0` or `-1` unless explicitly asked for (or if a crash occurs, but then the exit code would be a CLR HResult, at least `0x80000000`)

**Reply by mjr4077au** [reply]
It's definitely not a PSADT exit code (I've recently joined the development team for that also).

Further research seems to indicate it is a CLR HResult, which makes sense considering that's all I get from a transcription, indicating PowerShell was crashing.

However, it's also stopped happening which makes me think it was a transient Intune bug yesterday 😐. That's even more annoying as I was pulling my hair out wondering what was wrong.
